### ISSUE TITLE
wii flow r31 does not load cover images from fat partion on hdd when using WBFS, r25 does load them

### ISSUE THREAD

**Issue by GoogleCodeExporter** [WhiteNeo/wiiflow#125]
```
What steps will reproduce the problem?
1. Install wiiflow r31
2. Places cover images on the fat partion on hdd
3. Boot Wiiflow

What is the expected output? What do you see instead?

cover Images do not load from the fat partions on the hdd when using wbfs
Expecting images 

What version of the product are you using? On what operating system?

R31

```

Original issue reported on code.google.com by `samuelru...@gmail.com` on 6 Mar 2010 at 4:31

**Comment by GoogleCodeExporter** [comment]
```
I have a similar problem, can't get any covers to load in R31 workis in R25.

I have a 4Gb FAT partition on my USB disk, the rest is WBFS.

Folder sturcture is usb:\wiiflow\covers etc, I also tried moving the covers in 
to the
usb:\apps\wiiflow\covers and deleting config and cache, still no good.

Plasma
```

Original comment by `pla...@gmx.co.uk` on 8 Mar 2010 at 2:21

**Comment by GoogleCodeExporter** [comment]
```
make sure that you just have the wiiflow folder on SD/USB:\apss\wiiflow or 
SD/USB:
\wiiflow and delete the wiiflow.ini because the r28 and r31 looks for the 
wiiflow 
config in SD/USB:\wiiflow, if not found in SD/USB:\apss\wiiflow
```

Original comment by `naxox2...@gmail.com` on 8 Mar 2010 at 2:47

**Comment by GoogleCodeExporter** [comment]
```
I've tried that, I formatted the 4Gb Fat partition, and then re-created the
structure, copied over the r31 boot.dol, and the covers directory.. I deleted 
all the
.ini files and cleared the cache

usb:\apps\wiiflow\
usb:\apps\wiiflow\boxcovers
usb:\apps\wiiflow\cache
usb:\apps\wiiflow\covers
usb:\apps\wiiflow\themes

I'm wondering if the fact I'm booting it from PriiLoader has any bearing on 
this? R25
worked fine in this way. I don't use a forwarder (As I couldn't find one that
worked). So I have the r31 dol installed in PriiLoader as the "Installed File" 
as I
did with r25.

I also found I kept getting a "Hard Drive not found" with R31 on most loads, 
again
works sweet with R25. The only issue I have with r25 and why i was trying to 
update,
is the kids want to play Pepper Pig, and on launch it craches with a plain black
screen. I notice there is another issue for this already raised.

Plasma
```

Original comment by `pla...@gmx.co.uk` on 8 Mar 2010 at 2:56

**Comment by GoogleCodeExporter** [comment]
```
i have got the app on my SD, but trying to load the covers from the fat partion 
on 
hdd, not working with r31, but works perfect with r25
```

Original comment by `samuelru...@gmail.com` on 9 Mar 2010 at 4:32

**Comment by GoogleCodeExporter** [comment]
```
@samuelrusso6ge:look into the wiiflow.ini on your sd card and change the paths 
for
boxvocers and so on to the usb directory this should work fine.

@Plasma: Not really sure if priiloader is a problem since i never used it what
happens if you just use the HBC ?
```

Original comment by `thedarkn...@gmail.com` on 9 Mar 2010 at 1:55

**Comment by GoogleCodeExporter** [comment]
```
I'll try it tonight but it would seem the load behaviour has changed between 
r25 and
r31, I know I read the folders had moved to \apps\wiiflow from \wiiflow. But I 
tired
just about every configuration I could think of I even altered the wiiflow.ini 
file
manually. I'll maybe give it another go tonight via the HBC but that will mean
installing it on the SD or my USB disk.

PriiLoader installs the file in the NAND, so I'm not sure what the "home" 
directory
would be when it was run from there. But I'd be guessing the issue under 
PriiLoader
is that WiiFlow looks in the home/Launch directory for the wiiflow.ini, doesn't 
find
it and either tries to create one and fails or doesn't bother looking else 
where.
Maybe if it tried this list in some kind of order... 

usb:/apps/wiiflow/wiiflow.ini
usb:/wiiflow/wiiflow.ini
sd:/apps/wiiflow/wiiflow.ini
sd:/wiiflow/wiiflow.ini
LaunchDir/wiiflow.ini

R25 works fine with PriiLoader, I don't have the dol anywhere else, not on my 
FAT
partition or on my SD card, its loaded in to PriiLoader as the "installed" file 
and
thats it; it gets run as soon as I turn on my Wii. I just had the usb:/wiiflow/
structure and it all worked dandy.

Plasma
```

Original comment by `pla...@gmx.co.uk` on 9 Mar 2010 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
Ok, well I did some more testing and the outcome is rather strange ..
I have the CarbonicForwarder Channel installed on my wii. I made sure there was
nothing of WiiFlow in SD apps dir, and made sure the R31 dol at
usb:/apps/wiiflow/boot.dol was the only one present. With all my covers in the
usb:/apps/wiiflow/boxcovers etc. I deleted the Cache and the WiiFlow.ini files. 
So
this is a fresh clean run of R31.

Launched it from the Forwarder Channel and ....... no cover art, so I then 
remove the
SD and the UDB Disk to have a look what was happening.

R31 has created the following folder structure on my USB FAT Partition..

usb:/wiiflow
usb:/wiiflow/wiiflow.ini
usb:/wiiflow/boxcovers
usb:/wiiflow/cache
usb:/wiiflow/covers
usb:/wiiflow/themes

EVEN THOUGH it was run via a forwarder from usb:/apps/wiiflow/boot.dol

If I move my covers from usb:/apps/wiiflow/boxcovers to usb:/wiiflow/boxcovers 
and
restart from the forwarder channel I get my covers!! YEY! Though Pepper Pig 
still
doesn't work! :(

However, :( launching R31 from PriiLoader gives a WBFS_Open Failed: -1.

If I swap PriiLoader back to R25 it boots fine and loads the covers! I also, for
completeness, tried R27 this boots from PriiLoader but has no covers. R28 gives 
the
WBFS_Open Failed: -1 error just the same as R31. Please can this be rectified 
I'm
sure I'm not the only one using WiiFlow via PriiLoader. As it makes the wii 
interface
very swish and amazes all my mates!

Keep up the good work!

Plasma
```

Original comment by `pla...@gmx.co.uk` on 10 Mar 2010 at 12:33

**Comment by GoogleCodeExporter** [comment]
```
I've installed the rev31 on priiloader, and works fine to me, i mean i don't 
think 
that the priiloader is the problem...

Perhaps the difference it's that I have the wiiflow on my SD card and the games 
on a 
HDD (WBFS) and i have all the covers configurations and so on...

SD:\apps\wiiflow\
SD:\apps\wiiflow\boxcovers
SD:\apps\wiiflow\cache
SD:\apps\wiiflow\covers
SD:\apps\wiiflow\themes
SD:\r31.dol <---(priiloader)

```

Original comment by `naxox2...@gmail.com` on 10 Mar 2010 at 4:00

**Comment by GoogleCodeExporter** [comment]
```
Ok I might try it like that, I just liked the idea of having the Covers and the 
Games
on the same medium, so if I took it to a mates house it all went together 
without
having to remember to take the SD which might get lost when I'm drunk! ;)

Still be nice if we could mkae it work like in R25 as that would keep eveyone 
happy,
I'm comfortable shuffling things around and tweaking settings however some 
people
just don't get it.

I also noticed that Buttons 1 and 2 no longer swap the layout of the default 
theme,
is this intentional or is it a bug?

Plasma 

```

Original comment by `pla...@gmx.co.uk` on 11 Mar 2010 at 10:37

**Comment by GoogleCodeExporter** [comment]
```
Both these issues "PriiLoader" and Peppa Pig have now been fixed by r-win in a
release avaliable here .. I guess these fixes will make their way in to the main
branch shortly ...

http://www.mediafire.com/?qmtgtgjyynz

Thank you r-win!

Plasma
```

Original comment by `pla...@gmx.co.uk` on 15 Mar 2010 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
copying the wiiflow.ini from the hdd to SD does make the covers and themes load 
on
the latest r41 ios249

but i'll like it the way that it was in r25 so i dont need file duplicates 
everyone
```

Original comment by `samuelru...@gmail.com` on 9 Apr 2010 at 9:13

**Comment by GoogleCodeExporter** [comment]
```
I guess this issue is no issue anymore?
```

Original comment by `r-...@live.com` on 29 May 2010 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
Fixed in Wiiflow 2.0 or somewhere slightly before.
```

Original comment by `Miigotu` on 19 Jun 2010 at 10:42
- Changed state: **Fixed**
